pkg/operator/controller/ingress/metrics.go

```diff
@@ -68,6 +74,23 @@ func DeleteIngressControllerConditionsMetric(ic *operatorv1.IngressController) {
 	}
 }
 
+func DeleteActiveNLBMetrics(ic *operatorv1.IngressController) {
+	activeNLBs.DeleteLabelValues(ic.Name)
+}
+
+func SetIngressControllerNLBMetric(ci *operatorv1.IngressController) {
+	labelVal := 0
+	if ci.Status.EndpointPublishingStrategy != nil &&
+		ci.Status.EndpointPublishingStrategy.LoadBalancer != nil &&
+		ci.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil &&
+		ci.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.Type == operatorv1.AWSLoadBalancerProvider &&
+		ci.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS != nil &&
+		ci.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.Type == operatorv1.AWSNetworkLoadBalancer {
+		labelVal = 1
```

Comment:
There are better ways to express this, but I'll let it go for now. If you come back to this, can you clean this up ?

Reply:
I'm happy to apply any suggestion you have here